zebra/zebra_mlag.c

```diff
@@ -88,8 +878,557 @@ void zebra_mlag_init(void)
+					XMALLOC(MTYPE_MLAG_PBUF, vrf_name_len);
+
+				strlcpy(pay_load[i]->intf_name, msg.intf_name,
+					vrf_name_len);
+			}
+		}
+		if (cleanup == false) {
+			Bulk_msg.mroute_add = pay_load;
+			len = zebra_mlag_mroute_add_bulk__pack(&Bulk_msg,
+							       tmp_buf);
+		}
+
+		for (i = 0; i < mlag_msg.msg_cnt; i++) {
+			if (pay_load[i]->vrf_name)
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]->vrf_name);
+			if (pay_load[i]->owner_id == MLAG_OWNER_INTERFACE
+			    && pay_load[i]->intf_name)
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]->intf_name);
+			if (pay_load[i])
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]);
+		}
+		XFREE(MTYPE_MLAG_PBUF, pay_load);
+		if (cleanup == true)
+			return -1;
+	} break;
+	case MLAG_MROUTE_DEL_BULK: {
+		struct mlag_mroute_del msg;
+		ZebraMlagMrouteDelBulk Bulk_msg =
+			ZEBRA_MLAG_MROUTE_DEL_BULK__INIT;
+		ZebraMlagMrouteDel **pay_load = NULL;
+		int i = 0;
+		bool cleanup = false;
+
+		Bulk_msg.n_mroute_del = mlag_msg.msg_cnt;
+		pay_load = XMALLOC(MTYPE_MLAG_PBUF,
+				   sizeof(ZebraMlagMrouteDel *)
+					   * Bulk_msg.n_mroute_del);
+
+		for (i = 0; i < mlag_msg.msg_cnt; i++) {
+
+			uint32_t vrf_name_len = 0;
+
+			rc = zebra_mlag_lib_decode_mroute_del(s, &msg);
+			if (rc) {
+				cleanup = true;
+				break;
+			}
+
+			pay_load[i] = XMALLOC(MTYPE_MLAG_PBUF,
+					      sizeof(ZebraMlagMrouteDel));
+			zebra_mlag_mroute_del__init(pay_load[i]);
+
+			vrf_name_len = strlen(msg.vrf_name) + 1;
+			pay_load[i]->vrf_name =
+				XMALLOC(MTYPE_MLAG_PBUF, vrf_name_len);
+
+			strlcpy(pay_load[i]->vrf_name, msg.vrf_name,
+				vrf_name_len);
+			pay_load[i]->source_ip = msg.source_ip;
+			pay_load[i]->group_ip = msg.group_ip;
+			pay_load[i]->owner_id = msg.owner_id;
+			pay_load[i]->vrf_id = msg.vrf_id;
+			if (msg.owner_id == MLAG_OWNER_INTERFACE) {
+				vrf_name_len = strlen(msg.intf_name) + 1;
+				pay_load[i]->intf_name =
+					XMALLOC(MTYPE_MLAG_PBUF, vrf_name_len);
+
+				strlcpy(pay_load[i]->intf_name, msg.intf_name,
+					vrf_name_len);
+			}
+		}
+		if (cleanup == false) {
+			Bulk_msg.mroute_del = pay_load;
+			len = zebra_mlag_mroute_del_bulk__pack(&Bulk_msg,
+							       tmp_buf);
+		}
+
+		for (i = 0; i < mlag_msg.msg_cnt; i++) {
+			if (pay_load[i]->vrf_name)
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]->vrf_name);
+			if (pay_load[i]->owner_id == MLAG_OWNER_INTERFACE
+			    && pay_load[i]->intf_name)
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]->intf_name);
+			if (pay_load[i])
+				XFREE(MTYPE_MLAG_PBUF, pay_load[i]);
+		}
+		XFREE(MTYPE_MLAG_PBUF, pay_load);
+		if (cleanup == true)
+			return -1;
+	} break;
+	default:
+		break;
+	}
+
+	if (IS_ZEBRA_DEBUG_MLAG)
+		zlog_debug(
+			"%s: length of Mlag ProtoBuf encoded message:%s, %d",
+			__func__,
+			zebra_mlag_lib_msgid_to_str(mlag_msg.msg_type, buf, 80),
+			len);
+	hdr.type = (ZebraMlagHeader__MessageType)mlag_msg.msg_type;
+	if (len != 0) {
+		hdr.data.len = len;
+		hdr.data.data = XMALLOC(MTYPE_MLAG_PBUF, len);
+		if (hdr.data.data == NULL)
```

Comment:
I think the current practice is never to test return from the malloc wrappers - they will abort if out of memory.

Reply:
+1

Reply:
will take care